### PR TITLE
Initial CRUD attempt (WIP)

### PR DESCRIPTION
**Pull request by LiorB-D**

### CODE REVIEW

**Comment by LiorB-D** [comment]
@mdjorup Let me know if I'm in the right direction before I continue

File: package.json [hunk 1]
```diff
@@ -24,5 +24,8 @@
     "packageManager": "pnpm@8.12.1",
     "engines": {
         "node": ">=19"
+    },
+    "dependencies": {
+        "@supabase/supabase-js": "^2.39.1"
```
Comment (mdjorup) [inline package.json:29]: We're going to want to put this dependency in the `package.json` of the app or package that it's in. So For this one we would put it in `apps/web/package.json`

File: apps/web/src/util/supabaseClient.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export default function getSupabaseClient() {
+  const SUPABASE_API_URL = process.env.SUPABASE_API_URL;
+  const SUPABASE_ANON_KEY = process.env.SUPABASE_ANON_KEY;
+  if (!SUPABASE_API_URL) {
+    throw new Error("Missing Supabase API URL");
+  }
+  if (!SUPABASE_ANON_KEY) {
+    throw new Error("Missing Supabase ANON Key");
+  }
+  const supabase = createClient<Database>(SUPABASE_API_URL, SUPABASE_ANON_KEY);
```
Comment (mdjorup) [inline apps/web/src/util/supabaseClient.ts:12]: I'll have to look into how the `createClient` function works, but my understanding is that database connections are pretty expensive (as in they take a lot of time). Depending on the `createClient` implementation, we might be making a new connection on every call to `getSupabaseClient()`.
Either we should use https://www.npmjs.com/package/@supabase/auth-helpers-nextjs or we should make sure that the same connection is preserved on every call (probably by using some sort of singleton)

File: apps/web/src/app/api/advertisers/campaigns/[campaignId]/route.ts [hunk 3]
```diff
@@ -1,5 +1,38 @@
-import { type NextRequest } from "next/server";
+import { NextRequest, NextResponse } from "next/server";
+import getSupabaseClient from "../../../../../util/supabaseClient";
```
Comment (mdjorup) [inline apps/web/src/app/api/advertisers/campaigns/[campaignId]/route.ts:2]: I don't like the ridiculous amount of `../`. I will find a way to do import aliasing so it looks cleaner.

**Pull request closed without merging**